src/predicates.ts

```diff
@@ -576,37 +568,96 @@ export class AdminCommandContainerMissing implements Predicate<any> {
         container = command.replicated.container;
       }
 
-      const maybePath = this.checkCommand(root.components, component, container);
+      const matched = isContainerMissing(root.components, component, container);
 
-      if (maybePath) {
+      if (matched.matched) {
         paths.push(`admin_commands.${index}`);
-        paths.push(maybePath);
+      }
+
+      if (matched.paths) {
+        _.forEach(matched.paths, p => paths.push(p));
       }
     });
 
     return { matched: !_.isEmpty(paths), paths };
   }
 
-  private isKubernetesApp(commands: any[]) {
-    return !_.isEmpty(_.filter(commands, c => c.selector || c.selectors || (c.source && c.source.kuberentes)));
+}
+
+function isKubernetesApp(schedulerSourced: any[]) {
+  return !_.isEmpty(_.filter(schedulerSourced, c => c.selector || c.selectors || (c.source && c.source.kuberentes)));
+}
+
+function isSwarmApp(schedulerSourced: any[]) {
+  return !_.isEmpty(_.filter(schedulerSourced, c => c.service || (c.source && c.source.swarm)));
+}
+
+function isContainerMissing(components: any[], component: string, container: string): RuleMatchedAt {
+  // fail if component missing
+  const componentIndex: any = _.findIndex(components, { name: component });
+  if (componentIndex === -1) {
+    return { matched: true, paths: [`components`] };
```

Comment:
I can see how that could get confusing -- we've been using `matched: true` throughout the project to communicate that a rule was triggered and a warning should be shown.